src/analyses/cfg_dominators.h

```diff
@@ -26,20 +26,44 @@ template <class P, class T, bool post_dom>
 class cfg_dominators_templatet
 {
 public:
-  typedef std::set<T> target_sett;
+  typedef std::set<std::size_t> target_sett;
 
   struct nodet
   {
     target_sett dominators;
   };
 
-  typedef procedure_local_cfg_baset<nodet, P, T> cfgt;
+  typedef cfg_basic_blockst<procedure_local_cfg_baset, nodet, P, T> cfgt;
   cfgt cfg;
 
   void operator()(P &program);
 
   T entry_node;
 
+  /// Returns true if basic block \p lhs [post]dominates \p rhs
+  bool basic_block_dominates(std::size_t lhs, std::size_t rhs) const
+  {
+    return cfg[rhs].dominators.count(lhs);
+  }
+
+  /// Returns true if program point \p lhs [post]dominates \p rhs
+  bool dominates(T lhs, T rhs) const;
+
+  const target_sett &basic_block_dominators(std::size_t block) const
+  {
+    return cfg[block].dominators;
+  }
+
+  bool basic_block_reachable(std::size_t block) const
+  {
+    return !cfg[block].dominators.empty();
```

Comment:
Wouldn't this be false for the first block?

Reply:
All blocks dominate themselves by definition

Reply:
(except ones not reachable from the entry point, which are never visited by `cfg_dominatorst`)